Public/Src/Cache/ContentStore/Hashing/ContentHash.cs

```diff
@@ -214,6 +216,14 @@ public override string ToString()
             return Serialize();
         }
 
+        /// <summary>
+        /// Produces shorter string representation of the hash.
+        /// </summary>
+        public string ToShortString()
+        {
+            return new ShortHash(this).ToString();

```

Comment:
>new ShortHash(this) [](start = 19, length = 19)

It looks like we'll call this many times. Does it make sense to store `Lazy<ShortHash>` on the ContentHash?

Reply:
new ShortHash(this) won't allocate on the heap, but adding lazy field will increase the size of the ContentHash struct potentially causing perf regression.